wazo_dird/plugins/default_json/http.py

```diff
@@ -77,6 +78,57 @@ def get(self, profile):
         return response
 
 
+class LookupByUUID(LegacyAuthResource, DisplayAwareResource):
+    def __init__(
+        self, lookup_service, favorite_service, display_service, profile_service, auth_client
+    ):
+        self.lookup_service = lookup_service
+        self.favorite_service = favorite_service
+        self.display_service = display_service
+        self.profile_service = profile_service
+        self.auth_client = auth_client
+
+    @required_acl('dird.directories.lookup.{profile}.{xivo_user_uuid}.read')
+    def get(self, profile, xivo_user_uuid):
+        args = parser.parse_args()
+        term = args['term']
+
+        logger.info('Lookup %s for user %s with profile %s', term, xivo_user_uuid, profile)
+
+        try:
+            tenant_uuid = self._get_user_tenant_uuid(xivo_user_uuid)
```

Comment:
Using the tenant_uuid based on the user_uuid parameter allow a malicious user with the right ACL to do queries using the user_uuid of users in another tenant.

You should use the token's tenant_uuid or check that the user doing the query is above or in the user (from the parameter) tenant_uuid